Src/WindowsServer/WindowsServer.Shared/Implementation/AzureHeartbeatProperties.cs

```diff
@@ -0,0 +1,116 @@
+        /// <param name="resetCheckCompleteFlag">For testing: set to true to reset the check that we've already acquired this data.</param>
+        internal AzureHeartbeatProperties(IAzureMetadataRequestor azureInstanceMetadataHandler = null, bool resetCheckCompleteFlag = false)
+        {
+            this.azureInstanceMetadataRequestor = azureInstanceMetadataHandler;
+            if (this.azureInstanceMetadataRequestor == null)
+            {
+                this.azureInstanceMetadataRequestor = new AzureMetadataRequestor();
+            }
+
+            if (resetCheckCompleteFlag)
+            {
+                this.isAzureMetadataCheckCompleted = false;
+            }
+        }
+
+        /// <summary>
+        /// Add all enabled, present Azure IMS fields to the heartbeat properties.
+        /// </summary>
+        /// <param name="disabledFields">Fields that are to be disabled.</param>
+        /// <param name="provider">Heartbeat provider to set the properties on.</param>
+        /// <returns>True if any property values were successfully set, false if none were set.</returns>
+        public async Task<bool> SetDefaultPayload(IEnumerable<string> disabledFields, IHeartbeatPropertyManager provider)
+        {
+            bool hasSetFields = false;
+
+            if (!this.isAzureMetadataCheckCompleted)
+            {
+                this.isAzureMetadataCheckCompleted = true;
+
+                var azureComputeMetadata = await this.azureInstanceMetadataRequestor.GetAzureComputeMetadata()
+                                .ConfigureAwait(false);
+
+                if (azureComputeMetadata != null)
+                {
+                    var enabledImdsFields = this.ExpectedAzureImsFields.Except(disabledFields);
+                    foreach (string field in enabledImdsFields)
+                    {
+                        string value = azureComputeMetadata.GetValueForField(field);
+
+                        string verifiedValue = azureComputeMetadata.VerifyExpectedValue(field);
+
+                        bool addedProperty = provider.AddHeartbeatProperty(
+                                                        propertyName: string.Concat(AzureHeartbeatProperties.HeartbeatPropertyPrefix, field),
+                                                        propertyValue: verifiedValue,
+                                                        isHealthy: true);
+                        if (!addedProperty)
+                        {
+                            WindowsServerEventSource.Log.AzureInstanceMetadataWasntAddedToHeartbeatProperties(field, verifiedValue);
+                        }
+
+                        hasSetFields = hasSetFields || addedProperty;
+                    }
+                }
+                else
+                {
+                    WindowsServerEventSource.Log.AzureInstanceMetadataNotAdded();
+                }
+            }
+
+            return hasSetFields;
```

Comment:
where is this return value used?

Reply:
It is used in testing only, to assert that at least one value was added to the dictionary. It was also put there to follow best practices in returning from an async method.